### PR TITLE
Tests: For each test function, create an isolated project directory

### PR DESCRIPTION
**Pull request by hroncok**
This allows concurrent test execution.

Fixes https://github.com/fedora-python/tox-current-env/issues/11

### CODE REVIEW

**Comment by hroncok** [comment]
This works with xdist and detox. `tox --parallel` doesn't: https://github.com/tox-dev/tox/issues/1444

File: tests/test_integration.py [hunk 1]
```diff
@@ -13,18 +13,31 @@
 
 NATIVE_TOXENV = f"py{sys.version_info[0]}{sys.version_info[1]}"
 NATIVE_EXECUTABLE = str(pathlib.Path(sys.executable).resolve())
-TOX_INI = pathlib.Path(__file__).parent / "tox.ini"
-DOT_TOX = pathlib.Path(__file__).parent / ".tox"
+FIXTURES_DIR = pathlib.Path(__file__).parent / "fixtures"
```
Comment (torsava) [inline tests/test_integration.py]: I don't really understand this change. Why has the tox.ini moved from the same directory as `__file__` to the subdirectory `fixtures`?

Comment (torsava) [reply tests/test_integration.py]: As far as I can tell it's just a cosmetic change?

Comment (hroncok) [reply tests/test_integration.py]: Previously, the file was just there and it was used by the invoked tox via a config switch.
Now, it lives in a dedicated fixtures folder and is copied to a temporary directory, used from there.
You are right that this change was not strictly necesery, hence might be considered cosmetic. I usually put files used by pytest fixtures to a fixtures folder.

Comment (torsava) [reply tests/test_integration.py]: Ok, sounds good.

File: tests/test_integration.py [hunk 2]
```diff
@@ -246,11 +246,13 @@ def test_regular_after_first_deps_only_is_supported():
     assert f"/.tox/{NATIVE_TOXENV}/bin/python" in lines[0]
 
     # check that "test" was not installed to current environment
+    shutil.rmtree("./test.egg-info")
```
Comment (torsava) [inline tests/test_integration.py]: And here you don't care about the package being installed into the current dir? Or is just the egg-info being left over for some reason?

Comment (hroncok) [reply tests/test_integration.py]: the egg info in PWD fools pip freeze. it is not installed in current environment, it's just happens to be in the current dir.

Comment (torsava) [reply tests/test_integration.py]: Ah, ok.

File: tests/test_integration.py [hunk 3]
```diff
@@ -13,18 +13,31 @@
...
+    pwd.mkdir()
+    for fname in "tox.ini", "setup.py":
+        shutil.copy(FIXTURES_DIR / fname, pwd)
+
+    old_pwd = os.getcwd()
+    try:
+        os.chdir(pwd)
+        yield
+    finally:
+        os.chdir(old_pwd)
```
Comment (encukou) [inline tests/test_integration.py]: pytest has [`monkeypatch.chdir()`](https://docs.pytest.org/en/latest/monkeypatch.html) for this.

Comment (hroncok) [reply tests/test_integration.py]: added a fixup commit

**Comment by hroncok** [comment]
@torsava @encukou Thank You!

**Pull request merged**